Dockerfile

```diff
@@ -103,7 +108,11 @@ ENV JULIA_CPU_TARGET=x86_64;haswell;skylake;skylake-avx512;tigerlake
 WORKDIR "${REEFGUIDE_SRC_DIR}"
 COPY ./Project.toml ./Project.toml
 COPY ./Manifest.toml ./Manifest.toml
-RUN julia --project=@reefguide -e 'using Pkg;  Pkg.instantiate(verbose=true)'
+# Pre-download and cache MKL_jll dependency before general precompilation
+RUN julia -e 'using Pkg; Pkg.add("MKL_jll"); Pkg.instantiate(); using MKL_jll'
```

Comment:
Not sure which project is being instantiated here.

Additionally, are the two copy statements above necessary?

Am I correct in assuming the build is faster to complete because the step to precompile MKL and others is essentially skipped in subsequent builds?

Reply:
@ConnectedSystems  Yes, subsequent builds are faster because the MKL_jll and other dependencies are precompiled and cached, which prevents the need for recompilation at runtime.

Regarding the COPY statements: yes, they are needed to ensure Project.toml and Manifest.toml are available for Julia to install and precompile dependencies during Pkg.instantiate()

Reply:
> Regarding the COPY statements: yes, they are needed to ensure Project.toml and Manifest.toml are available for Julia to install and precompile dependencies during Pkg.instantiate()

Yes, I'm asking for clarification as it looks like it is redundant (copy to same location?)

```
COPY ./here ./here
```

Reply:
@ConnectedSystems  No, it was not redundant. COPY ./Project.toml ./Project.toml transfers Project.toml from the local directory into the container, which is necessary for dependency setup. I have simplified this further by consolidating it into a single line: COPY Project.toml Manifest.toml ./.

Reply:
Thanks for the clarification. Obviously I'm not so familiar with Docker